libs/community/langchain_community/agent_toolkits/sql/base.py

```diff
@@ -140,8 +141,9 @@ def create_sql_agent(
     toolkit = toolkit or SQLDatabaseToolkit(llm=llm, db=db)  # type: ignore[arg-type]
     agent_type = agent_type or AgentType.ZERO_SHOT_REACT_DESCRIPTION
     tools = toolkit.get_tools() + list(extra_tools)
+    if prefix is None:
+        prefix = SQL_PREFIX
```

Comment:
why move this out? we should only have a prefix if `prompt is None`, no?

Reply:
This was the original  behaviour as shown in the line [here](https://github.com/keenborder786/langchain/blob/0e916d0d552fde282fa3bf51010c47536913bba3/libs/community/langchain_community/agent_toolkits/sql/base.py#L143). If you want I can change it.

Reply:
Oh I see your point. It has been reverted.